server/sublist.go

```diff
@@ -295,10 +319,18 @@ func (s *Sublist) Match(subject string) *SublistResult {
 
 	// Get result from the main structure and place into the shared cache.
 	// Hold the read lock to avoid race between match and store.
+	var n int32
+
 	s.RLock()
 	matchLevel(s.root, tokens, result)
```

Comment:
I think that to completely avoid the mem allocation (`result := &SublistResult{}`), we should have `result = matchLevel(s.root, tokens, result)` and allocate a `&SublistResult{}` in `matchLevel()` when applicable. Then you could check that if returned `result` is nil, then assign it `emptyResult`.
We have discussed that and you think that it is ok this way for now.

Reply:
I think escape analysis may be good enough now to know the stack one does not escape when result is over written. Or if not now maybe soon. Agree ok for now.